src/registrar/models/utility/domain_helper.py

```diff
@@ -29,19 +29,19 @@ def validate(cls, domain: str | None, blank_ok=False) -> str:
         if not isinstance(domain, str):
             raise ValueError("Domain name must be a string")
         domain = domain.lower().strip()
-        if domain == "":
-            if blank_ok:
-                return domain
-            else:
-                raise errors.BlankValueError()
+        if domain == "" and not blank_ok:
+            raise errors.BlankValueError()
         if domain.endswith(".gov"):
             domain = domain[:-4]
         if "." in domain:
             raise errors.ExtraDotsError()
         if not DomainHelper.string_could_be_domain(domain + ".gov"):
             raise ValueError()
-        if not check_domain_available(domain):
-            raise errors.DomainUnavailableError()
+        try:
+            if not check_domain_available(domain):
+                raise errors.DomainUnavailableError()
```

Comment:
(Q) was line 41 throwing an exception?

Reply:
I believe so! I think lines 41-42 were the original code and we put it in a try catch since we're now throwing a RegistryError when the availability API fails